homeassistant/components/energy/sensor.py

```diff
@@ -223,6 +224,16 @@ def _update_cost(self) -> None:
         if energy_state is None:
             return
 
+        if (
+            state_class := energy_state.attributes.get(ATTR_STATE_CLASS)
+        ) != STATE_CLASS_TOTAL_INCREASING:
+            _LOGGER.warning(
```

Comment:
Do we need to limit this log to avoid spamming users?

Reply:
Yes, that makes sense.
Fixed in cc28389